File: Caesar Cipher Encryptor/O(n)_time_and_space.js
function caesarCipherEncryptor(string, key) {
  // Write your code here.
  let newLetters = [];
  key %= 26;
  for (let char of string) {
    newLetters.push(getNewLetter(char, key));
  }
  return newLetters.join("");
}

function getNewLetter(letter, key) {
  let newLetterCode = letter.charCodeAt() + key;
  return newLetterCode <= 122
    ? String.fromCharCode(newLetterCode)
    : String.fromCharCode(96 + newLetterCode);
}

File: Strings/Caesar Cipher Encryptor/O(n)_time_and_space.py
def caesarCipherEncryptor(string, key):
    # Write your code here.
    newLetters=[]
    key=key%26
    for char in string:
        newLetters.append(getNewLetter(char,key))

    return "".join(newLetters)

def getNewLetter(letter,key):
    newLetterCode = ord(letter) + key
    return chr(newLetterCode) if newLetterCode<=122 else chr(96 +(newLetterCode%122))